### PR TITLE
fix dataframe only boolean/binary column got error on describe

### LINKED ISSUE #5558: Dataframe describe() method cannot handle dataframes without a numeric type column
**Describe the bug**

If calling `describe(...)` on a dataframe without a numeric type method then it returns an error.

**To Reproduce**

```rust
    ctx.sql("select 'a' as a")
        .await?
        .describe()
        .await?
        .show()
        .await?;
```

Causes error:

```
Error: Plan("Aggregate requires at least one grouping or aggregate expression")
```

**Expected behavior**

Either be able to handle this properly or return a more informative error

**Additional context**

Caused by filters for certain aggregations:

https://github.com/apache/arrow-datafusion/blob/ecbc843a1a8c38b2466748bc92a6e22ce08d51ed/datafusion/core/src/dataframe.rs#L371-L449

- See mean, std, median

Also worth testing for if the input dataframe has only boolean boolean/binary column, like so:

```rust
    ctx.sql("select true as a")
        .await?
        .describe()
        .await?
        .show()
        .await?;
```

- See max & min in above code snippet

### PR DESCRIPTION
**Pull request by jiangzhx**
# Which issue does this PR close?

Closes #5558

- handle input dataframe has only boolean/binary column got error
- add more test

### CODE REVIEW

File: datafusion/core/src/dataframe.rs [hunk 1]
```diff
@@ -437,19 +416,27 @@ impl DataFrame {
...
+                                array_datas
+                                    .push(Arc::new(StringArray::from_slice(["null"])));
+                            }
                         }
                     }
-                    //if None mean the column cannot be min/max aggregation
-                    None => {
+                    Err(_) => {
+                        //Handling error when only boolean/binary column, and in other cases
                         array_datas.push(Arc::new(StringArray::from_slice(["null"])));
```
Comment (jiangzhx) [inline datafusion/core/src/dataframe.rs:440]: Maybe it's not a good idea to catch all errors and return StringArray::from_slice(["null"]).

Comment (alamb) [reply datafusion/core/src/dataframe.rs:440]: Maybe you can match on the error message 

Something like 

```rust
Err(e) if e.to_string().contains("message") => {
  ...
}
_ => {
 // Default error
}
```

File: datafusion/core/src/dataframe.rs [hunk 2]
```diff
@@ -437,19 +416,27 @@ impl DataFrame {
...
-                    Some(c) => {
-                        if field.data_type().is_numeric() {
-                            array_datas.push(cast(c, &DataType::Float64)?);
-                        } else {
-                            array_datas.push(cast(c, &DataType::Utf8)?);
+            for result in describe_record_batch.iter() {
+                match result {
+                    Ok(df) => {
+                        let record_batch = df.clone().collect().await?;
+                        if record_batch.len() == 1 {
```
Comment (comphead) [inline datafusion/core/src/dataframe.rs:423]: Thiking, if the code can be simplified a bit, now the code looks overloaded with conditions.

Comment (jiangzhx) [reply datafusion/core/src/dataframe.rs]: Thanks, I made some changes. If you have time, could you please help with another code review?

Comment (jiangzhx) [reply datafusion/core/src/dataframe.rs]: > Thiking, if the code can be simplified a bit, now the code looks overloaded with conditions.

Thanks, I made some changes. If you have time, could you please help with another code review?

**Review by alamb — APPROVED** [review]
Thanks @jiangzhx  -- while I agree with @comphead  that the code seems like it could be improved, I think the test coverage is good and we can refine the code as a follow on PR.

Thank you for this contribution

**Pull request merged**